### PR TITLE
QAP-109 Eldorado e2e buying mobile phone;

### PR DESCRIPTION
**Pull request by YevheniiBond**

### CODE REVIEW

File: src/test/java/YevheniiBondarenko/Tests/Pages/EldoradoWebElementsPage.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    }
+    private static final String SEARCH_FIELD = "//input[@class='search-input']";
+    private static final String MOBILE_PHONES = "//li[@class='cat-title']/child::a[contains(text(),'Мобільні телефони')]";
+    private static final String BROVARY_CITY = "//div[@id='input-66']";
+    private static final String LIST_OF_MOBILE_PHONES = "//div[@class='goods-item-content']";
+    private static final String COLOR_OF_THE_PHONE = "//ul[@class='product-color-list product-page-colors-list']/child::li";
+    private static final String BUY_BUTTON = "(//div[contains(@class,'buy-button-placeholder')]/div)[2]";
+    private static final String CART_ITEMS_NUMBER = "//div[@class='counter counter-cart active']";
+    private static final String DELETE_ITEMS_FROM_CART = "//*[name()='svg']";
+
```
Comment (Wolmeright) [inline src/test/java/YevheniiBondarenko/Tests/Pages/EldoradoWebElementsPage.java:20]: Delete empty line, please)

File: src/test/java/YevheniiBondarenko/Tests/Pages/EldoradoWebElementsPage.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    public WebElement searchingField(){return $(SEARCH_FIELD);}
+    public WebElement mobilePhones(){return  $(MOBILE_PHONES);}
+    public WebElement brovaryCity(){return $(BROVARY_CITY);}
+    public WebElement cartItemsNumber(){return $(CART_ITEMS_NUMBER);}
+    public List<WebElement> listOfMobilePhones(){return $$(LIST_OF_MOBILE_PHONES);}
+    public List<WebElement> colorOfMobilePhones(){return $$(COLOR_OF_THE_PHONE);}
+    public WebElement buyButton(){return $(BUY_BUTTON);}
+    public List<WebElement> deleteItemsFromCart(){return $$(DELETE_ITEMS_FROM_CART);}
+
+
```
Comment (Wolmeright) [inline src/test/java/YevheniiBondarenko/Tests/Pages/EldoradoWebElementsPage.java:31]: Delete empty lines, please)

File: src/test/java/YevheniiBondarenko/Tests/Tests/EldoradoE2EBuyingMobilePhone.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        eldoradoWebElementsPage.deleteItemsFromCart().get(0).click();
+
+        Assert.assertTrue(eldoradoWebElementsPage.cartItemsNumber().isEnabled());
+
+
+
+
+
+
+
```
Comment (Wolmeright) [inline src/test/java/YevheniiBondarenko/Tests/Tests/EldoradoE2EBuyingMobilePhone.java:37]: Delete empty lines, please)

**Review by Wolmeright — CHANGES_REQUESTED** [review]

**Review by YevheniiBond — COMMENTED** [review]
Done

File: src/test/java/YevheniiBondarenko/Tests/Tests/EldoradoE2EBuyingMobilePhone.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import YevheniiBondarenko.Tests.Pages.EldoradoWebElementsPage;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.interactions.Actions;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class EldoradoE2EBuyingMobilePhone extends TestInit {
+
+    @Test
+    public void eldoradoE2EAddAndRemoveFromCart(){
```
Comment (AndriiMyronenko418) [inline src/test/java/YevheniiBondarenko/Tests/Tests/EldoradoE2EBuyingMobilePhone.java:13]: add an empty line between 13 and 14

**Review by AndriiMyronenko418 — CHANGES_REQUESTED** [review]

File: src/test/java/YevheniiBondarenko/Tests/Pages/EldoradoWebElementsPage.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    private static final String DELETE_ITEMS_FROM_CART = "//*[name()='svg']";
+
+    public WebElement searchingField(){return $(SEARCH_FIELD);}
+    public WebElement mobilePhones(){return  $(MOBILE_PHONES);}
+    public WebElement brovaryCity(){return $(BROVARY_CITY);}
+    public WebElement cartItemsNumber(){return $(CART_ITEMS_NUMBER);}
+    public List<WebElement> listOfMobilePhones(){return $$(LIST_OF_MOBILE_PHONES);}
+    public List<WebElement> colorOfMobilePhones(){return $$(COLOR_OF_THE_PHONE);}
+    public WebElement buyButton(){return $(BUY_BUTTON);}
+    public List<WebElement> deleteItemsFromCart(){return $$(DELETE_ITEMS_FROM_CART);}
```
[the remark this answers is not in the document]
Comment (Wolmeright) [reply src/test/java/YevheniiBondarenko/Tests/Pages/EldoradoWebElementsPage.java:28]: use CTRL+ALT+L

**Review by Wolmeright — APPROVED** [review]

**Review by Tsibatyuk — APPROVED** [review]

File: src/test/java/YevheniiBondarenko/Tests/Pages/EldoradoWebElementsPage.java (hunk 1 above)
Comment (YevheniiBond) [reply src/test/java/YevheniiBondarenko/Tests/Pages/EldoradoWebElementsPage.java:20]: +

File: src/test/java/YevheniiBondarenko/Tests/Pages/EldoradoWebElementsPage.java (hunk 2 above)
Comment (YevheniiBond) [reply src/test/java/YevheniiBondarenko/Tests/Pages/EldoradoWebElementsPage.java]: +++

File: src/test/java/YevheniiBondarenko/Tests/Pages/EldoradoWebElementsPage.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (YevheniiBond) [reply src/test/java/YevheniiBondarenko/Tests/Pages/EldoradoWebElementsPage.java:28]: +++

File: src/test/java/YevheniiBondarenko/Tests/Tests/EldoradoE2EBuyingMobilePhone.java (hunk 4 above)
Comment (YevheniiBond) [reply src/test/java/YevheniiBondarenko/Tests/Tests/EldoradoE2EBuyingMobilePhone.java:13]: +++

File: src/test/java/YevheniiBondarenko/Tests/Tests/EldoradoE2EBuyingMobilePhone.java (hunk 3 above)
Comment (YevheniiBond) [reply src/test/java/YevheniiBondarenko/Tests/Tests/EldoradoE2EBuyingMobilePhone.java]: +++

**Review by AndriiMyronenko418 — APPROVED** [review]

**Pull request merged**